src/lib.rs

```diff
@@ -18,11 +19,44 @@ use walkdir::WalkDir;
 pub struct Vessel {
     pub package_set: PackageSet,
     pub manifest: Manifest,
+    /// How many parent directories are we nested underneath the project root
+    pub nested: u32,
 }
 
 impl Vessel {
+    fn find_dominating_manifest() -> Result<Option<u32>> {
+        let mut cwd = env::current_dir().context("Unable to access the current directory")?;
+        let mut nested = 0;
+        loop {
+            if cwd.join("vessel.dhall").exists() {
+                if nested != 0 {
+                    info!("Changing working directory to {}", cwd.display());
+                    env::set_current_dir(&cwd).context(format!(
+                        "Failed to change current directory to {}",
```

Comment:
this will construct to error message even if unused.

Reply:
I'm not too worried about that, this is run exactly once per invocation.